### DISCUSSION TITLE
Defense Bots should be able to move upward

### DISCUSSION THREAD

**Discussion by G-M-twostay** [FakeFishGames/Barotrauma#12207] [Ideas and suggestions]
Despite defense bots float in air using propellers like toy helicopters, they can't move upwards or use ladders. This can cause big issues. We know from the wiki that defense bots move around randomly; we also know that it effectively can't move upwards in a sub where ladders are the only vertical movement method(very common, like dugong). This forms a random walk that cannot go up. By the property of random walk, the defense bots will always eventually end up the bottom of the sub, which is usually the ballast, and be stuck there. This limits their functionality a lot. 

The description of defense bots says that they're designed to roam the sub and shoot at hostiles, or protect the mechanic that built them. First, if they r constantly stuck in ballast, then they probably can't protect much of the sub from monsters. I've never seen a sub where there is this single very large flat ballast tank that cover most of the sub. Second, because no human on sub tends to stay in the ballast, the bots stuck in ballast effectively means that they can't protect the mechanic, completely opposite of what's intended.

Take a step back, even if they aren't in the ballast, they still would be in the bottom layer of the sub, this still limits their usages.

In conclusion, giving defense bots the ability to move upwards would be a very huge buff to them, as well as making them behave more consistently and meet expectations(who would expect something that floats in air can't move up). They don't have to actually use ladders or follow waypoints, as they are bots that uses propellers and not legs. It can be made that they will just move up in the sub, not necessarily using ladders. Also, they should be able to stand on platforms, like everything else.

**Comment by G-M-twostay** [comment]
Pets should also have a way to move upward. My orange boy alwasy ends up in ballast.

**Reply by Hazmat616** [reply]
They should call for help so even bots will grab them and move them up ladders or stairs

**Comment by Hazmat616** [comment]
Defense bot should have the equivalent to a dive scooter for in the air. So it can just freely move

**Comment by itchyOwl** [comment]
Afaik there was a technical issue why it couldn't use the ladders (maybe because they aren't humanoids), so it was disabled. But I agree we should address this. Creating a ticket: #14321.

**Comment by Regalis11** [comment]
Duplicate of https://github.com/FakeFishGames/Barotrauma/issues/11793
